README.md

```diff
@@ -1,2 +1,113 @@
 # HSM-SDS-Server
-HSM SDS Server for private key protection in Envoy mTLS and Istio Gateway.
+
+## Introduction
+
+The HSM SDS Server follows the SDS extension standard of Envoy and implements an external SDS server via more secure solution which is known as Hardware Security Module(HSM). By using this repo, User can maintain the credentials for workloads managed by Istio/Envoy in more secure scenario via external SDS server Besides supporting management for new credentials, it also allows user to upload existing credentials and manages them in higher security level. This external SDS server can be used for both gateways and workload sidecars to provide their credential information.
+
+This HSM SDS Server protects service mesh data plane private keys with Intel® SGX. The private keys are stored and used inside the SGX enclave(s) and will never stored in clear anywhere in the system. Authorized applications use the private key in the enclave by key-handle provided by SGX.
+
+## Architecture
+
+### Workload mTLS private key protection:
+<img src="diagrams/mTLS-arch.png">
+
+### Gateway private key protection:
+TODO: Add gateway arch diagram.
+## Prerequisites
+
+Prerequisites for using Istio mTLS private key protection with SGX:
+
+- Kubernetes cluster with one or more nodes with Intel® [SGX](https://software.intel.com/content/www/us/en/develop/topics/software-guard-extensions.html) supported hardware
+- [Intel® SGX device plugin for Kubernetes](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/cmd/sgx_plugin/README.md)
+- [Intel® SGX AESM daemon](https://github.com/intel/linux-sgx#install-the-intelr-sgx-psw)
+- [Intel® KMRA service](https://www.intel.com/content/www/us/en/developer/topic-technology/open/key-management-reference-application/overview.html)
+- Linux kernel version 5.11 or later on the host (in tree SGX driver)
+- [trusted-certificate-issuer](https://github.com/intel/trusted-certificate-issuer)
```

Comment:
Can we make this as an optional dependency?

Reply:
AESM and KMRA are not need if `NEED_QUOTE` disabled, otherwise these all needed.